### ISSUE TITLE
Production mode with sourcemaps (Debug Build)

### ISSUE THREAD

**Issue by Shard** [liquid-state/generator-ls-iwa#7]
Need to have a clean way to build apps for production with sourcemaps and without instead of having to edit webpack config with some of the overrides. What I propose is spliting the builds into `production` and `debug`.

`production` works as currently, proper optimized build and `debug` would build with the production ENV variable on and everything else that should be on so there is no dev mode mocking but still include source maps and potentially things like redux dev tools.

**Comment by elsimir** [comment]
@Shard I disagree with making build with maps the default. While it can be useful in development the size of the source maps is not inconsequential and we should make sure people don't bloat their app by accident. Debugging with maps on device is not the most common use case especially as we iron out more issues.

**Comment by Shard** [comment]
Thanks @elsimir, Updated to reflect.

**Comment by cyrildoussin** [comment]
FWIW it's worth I agree with @elsimir Source maps are too large which causes a few problems:

- downloaded IWAs are too big (in time to download and in time to decompress)
- app builds are significantly larger when IWAs are built in
- large bundles generally (and possibly particularly source maps) are a likely cause for issues like the Safari inspector crashing when debugging on device.

Therefore, I don't think it's reasonable to leave them in builds by default.
